mm/ktsan/Makefile

```diff
@@ -1,28 +1,70 @@
-obj-y := ktsan.o		\
-	 thr.o			\
-	 shadow.o		\
-	 access.o		\
-	 memblock.o		\
-	 sync.o			\
-	 sync_mtx.o		\
-	 sync_atomic.o		\
-	 sync_percpu.o		\
-	 clk.o			\
-	 tab.o			\
-	 report.o		\
-	 stat.o			\
-	 alloc.o		\
-	 atomic_no_ktsan.o	\
-	 stack.o		\
-	 trace.o		\
-	 func.o			\
-	 tests_noinst.o		\
-	 tests_inst.o		\
-	 tests.o
+obj-y := access.o 		\
+	 alloc.o 		\
+	 atomic_no_ktsan.o 	\
+	 clk.o 			\
+	 func.o 		\
+	 ktsan.o 		\
+	 memblock.o 		\
+	 report.o 		\
+	 shadow.o 		\
+	 stack.o 		\
+	 stat.o 		\
+	 sync_atomic.o 		\
+	 sync_mtx.o 		\
+	 sync.o 		\
+	 sync_percpu.o 		\
+	 tab.o 			\
+	 tests_inst.o 		\
+	 tests_noinst.o 	\
+	 tests.o 		\
+	 thr.o 			\
+	 trace.o 		\
 
-CFLAGS_REMOVE_ktsan.o = -pg -O0 -O1 -O2
-CFLAGS_ktsan.o = -O3
+CFLAGS_REMOVE_access.o		= -pg -O0 -O1 -O2
+CFLAGS_REMOVE_alloc.o		= -pg -O0 -O1 -O2
+CFLAGS_REMOVE_atomic_no_ktsan.o	= -pg -O0 -O1 -O2
+CFLAGS_REMOVE_built-in.o	= -pg -O0 -O1 -O2
+CFLAGS_REMOVE_clk.o		= -pg -O0 -O1 -O2
+CFLAGS_REMOVE_func.o		= -pg -O0 -O1 -O2
+CFLAGS_REMOVE_ktsan.o		= -pg -O0 -O1 -O2
+CFLAGS_REMOVE_memblock.o	= -pg -O0 -O1 -O2
+CFLAGS_REMOVE_report.o		= -pg -O0 -O1 -O2
+CFLAGS_REMOVE_shadow.o		= -pg -O0 -O1 -O2
+CFLAGS_REMOVE_stack.o		= -pg -O0 -O1 -O2
+CFLAGS_REMOVE_stat.o		= -pg -O0 -O1 -O2
+CFLAGS_REMOVE_sync_atomic.o	= -pg -O0 -O1 -O2
+CFLAGS_REMOVE_sync_mtx.o	= -pg -O0 -O1 -O2
+CFLAGS_REMOVE_sync.o		= -pg -O0 -O1 -O2
+CFLAGS_REMOVE_sync_percpu.o	= -pg -O0 -O1 -O2
+CFLAGS_REMOVE_tab.o		= -pg -O0 -O1 -O2
+CFLAGS_REMOVE_tests_inst.o	= -pg -O0 -O1 -O2
+CFLAGS_REMOVE_tests_noinst.o	= -pg -O0 -O1 -O2
+CFLAGS_REMOVE_tests.o		= -pg -O0 -O1 -O2
+CFLAGS_REMOVE_thr.o		= -pg -O0 -O1 -O2
+CFLAGS_REMOVE_trace.o		= -pg -O0 -O1 -O2
 
-KTSAN_SANITIZE := n
+CFLAGS_access.o			= -O3
```

Comment:
I think there must be a macro that adds a flag for all files in a dir.
KBUILD_CFLAGS may be that macro.
I did not find anything similar for flag removal.